app/src/test/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavViewModelTest.kt

```diff
@@ -1333,6 +1334,45 @@ class RootNavViewModelTest : BaseViewModelTest() {
         )
     }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `when the active user has an unlocked vault and they need to be shown the new device notice the nav state should be NewDeviceTwoFactorNotice`() {
+        every { authRepository.checkUserNeedsNewDeviceTwoFactorNotice() } returns true
+        mutableUserStateFlow.tryEmit(
```

Comment:
🤔 I think another test should be added where `authRepository.checkUserNeedsNewDeviceTwoFactorNotice` returns `false` to make sure the navigation goes to the proper place as well.

Reply:
yup makes sense